kong/db/dao/snis.lua

```diff
@@ -85,18 +85,26 @@ end
 
 -- Returns the name list for a given certificate
 function _SNIs:list_for_certificate(cert_pk, options)
-  local name_list = setmetatable({}, cjson.empty_array_mt)
+  local name_list = {}
+  local name_count = 0
 
   for sni, err, err_t in self:each_for_certificate(cert_pk, 1000, options) do
     if err then
       return nil, err, err_t
     end
 
     table.insert(name_list, sni.name)
+    name_count = name_count + 1
   end
 
   table.sort(name_list)
 
+  if name_count == 0 then
+    setmetatable(name_list, cjson.empty_array_mt)
+  else
+    setmetatable(name_list, cjson.array_mt)
```

Comment:
You can forget about `cjson.empty_array_mt` and simply stick to setting `array_mt` unconditionally (this could just be a one line change).